pkg/sparrow/targets/targetmanager.go

```diff
@@ -0,0 +1,38 @@
+// sparrow
+// (C) 2023, Deutsche Telekom IT GmbH
+//
+// Deutsche Telekom IT GmbH and all other contributors /
+// copyright owners license this file to you under the Apache
+// License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package targets
+
+import (
+	"context"
+
+	"github.com/caas-team/sparrow/pkg/checks"
+)
+
+// TargetManager handles the management of globalTargets for
+// a Sparrow instance
+type TargetManager interface {
+	// Reconcile fetches the global targets from the configured
+	// endpoint and updates the local state
+	Reconcile(ctx context.Context)
+	// GetTargets returns the current global targets
+	GetTargets() []checks.GlobalTarget
+	// Shutdown shuts down the target manager
+	// and unregisters the instance as a global target
+	Shutdown(ctx context.Context) error
```

Comment:
You're returning an error here, kind of like we do in the `Checks` interface. Was that on purpose to keep things consistent? 

In the current implementation you're always returning `nil` though, so I'm wondering if we even need the error return here?